cgo/lapack.go

```diff
@@ -413,6 +425,63 @@ func (impl Implementation) Dlarfx(side blas.Side, m, n int, v []float64, tau flo
 	lapacke.Dlarfx(side, m, n, v, tau, c, ldc, work)
 }
 
+// Dlascl multiplies an m×n matrix by the scalar cto/cfrom.
+//
+// cfrom must not be zero, and cto and cfrom must not be NaN, otherwise Dlascl
+// will panic.
+//
+// Dlascl is an internal routine. It is exported for testing purposes.
+func (impl Implementation) Dlascl(kind lapack.MatrixType, kl, ku int, cfrom, cto float64, m, n int, a []float64, lda int) {
+	checkMatrix(m, n, a, lda)
+	if cfrom == 0 {
+		panic(zeroDiv)
+	}
+	if math.IsNaN(cfrom) || math.IsNaN(cto) {
+		panic(nanScale)
+	}
+	lapacke.Dlascl(byte(kind), kl, ku, cfrom, cto, m, n, a, lda)
+}
+
+// Dlaset sets the off-diagonal elements of A to alpha, and the diagonal
```

Comment:
This doesn't currently match the behaviour of lapacke.Dlaset which panics is uplo is not blas.Upper or blas.Lower. To fix this, you need to add the following and regenerate lapacke:

```diff
diff --git a/cgo/lapacke/generate_lapacke.go b/cgo/lapacke/generate_lapacke.go
index 9110c9a..02432c1 100644
--- a/cgo/lapacke/generate_lapacke.go
+++ b/cgo/lapacke/generate_lapacke.go
@@ -64,6 +64,7 @@ var needsInt = map[string]bool{
 // The list keys are truncated by one character to cover all four numeric types.
 var allUplo = map[string]bool{
        "lacpy": true,
+       "laset": true,
 }
```

Note that even then, the comment is not true. The uplo check checks for U, L, A and panics if it is not one of those. This could be relaxed I guess - ?lacpy has the same horrible behaviour.